compiler/rustc_middle/src/ty/fold.rs

```diff
@@ -471,9 +471,12 @@ impl<'a, 'tcx> TypeFolder<'tcx> for BoundVarReplacer<'a, 'tcx> {
         match *t.kind() {
```

Comment:
I wonder whether something like this would be more readable:
```rust
fn fold_ty(&mut self, t: Ty<'tcx>) -> Ty<'tcx> {
    match *t.kind() {
        ty::Bound(debruijn, bound_ty) if debruijn == self.current_index => {
            if let Some(fld_t) = self.fld_t.as_mut() {
                let ty = fld_t(bound_ty);
                return ty::fold::shift_vars(self.tcx, &ty, self.current_index.as_u32());
            }
        }
        _ if t.has_vars_bound_at_or_above(self.current_index) => return t.super_fold_with(self),
    }
    t
}
```
(and similarly for the other methods).

Reply:
Ooh, I like that